configure.ac

```
@@ -45,6 +45,15 @@ AS_IF([test "x$enable_analyzer" = "xyes"], [
 	EXTRA_CFLAGS="$EXTRA_CFLAGS -fanalyzer -Wno-analyzer-malloc-leak"
 ])
 
+AC_ARG_ENABLE([sanitizer],
+    AS_HELP_STRING([--enable-sanitizer=@<:@address,memory,undefined@:>@], [enable address/memory/undefined sanitizer (debug)]))
+AS_IF([test "x$enable_sanitizer" = "xaddress" -o "x$enable_sanitizer" = "xmemory" -o "x$enable_sanitizer" = "xundefined"],
+	[AX_CHECK_COMPILE_FLAG([-fsanitize=$enable_sanitizer],[
+		EXTRA_CFLAGS="$EXTRA_CFLAGS -fsanitize=$enable_sanitizer -fno-omit-frame-pointer"
+		EXTRA_LDFLAGS="$EXTRA_LDFLAGS -fsanitize=$enable_sanitizer"
+],[AC_MSG_ERROR([$enable_sanitizer sanitizer not supported])]
+)])
+
```

```suggestion
AC_ARG_ENABLE([sanitizer],
    AS_HELP_STRING([--enable-sanitizer=@<:@address | memory | undefined@:>@],
        [enable a compiler-based sanitizer (debug)]), [], [enable_sanitizer=no])
AS_IF([test "x$enable_sanitizer" != "xno"], [
	AX_CHECK_COMPILE_FLAG([-fsanitize=$enable_sanitizer], [
		EXTRA_CFLAGS="$EXTRA_CFLAGS -fsanitize=$enable_sanitizer -fno-omit-frame-pointer"
		EXTRA_LDFLAGS="$EXTRA_LDFLAGS -fsanitize=$enable_sanitizer"
	], [AC_MSG_ERROR([sanitizer not supported: $enable_sanitizer])])
])

```

Main suggestion:

* Just test if flag was passed and then let `AX_CHECK_COMPILE_FLAG` deal with
  it (so that there's less need to maintain the sanitizer list)

Nitpicks:

* Make help string more cli-like (with pipes == OR)
* Make error message more greppable/more like perror(3p) messages (static part
  first, etc)